### DISCUSSION TITLE
How to set primary billing & shipping address?

### DISCUSSION THREAD

**Discussion by johndwells** [craftcms/commerce#3304] [Q&A] [answered]
I haven't found any documentation on the Commerce docs regarding how to set a customer's primary billing & shipping addresses. I can see from the example templates that there is an ability during checkout to add an address and flag it to be one/both of these primary addresses, but in terms of _editing_ this preference later, I can't see any evidence as to how to accomplish this.

What I have managed to do is, if I add a custom field to the User's profile, with a handle of `primaryShippingAddressId`, if I set that value to a valid address ID and then submit a form to `users/save-user`, this value does appear to be captured and stored in the Customers table too. So I've managed to get it to work, but it seems almost by accident.

If I submit a form to `users/save-user` with a field `primaryShippingAddressId`, but there isn't a custom field of the same handle, it does not get passed to the Customer's table.

Same applies for `primaryBillingAddressId` too.

Looking forward to any input!

**Comment by lukeholder** [comment, marked as answer]
You would submit `isPrimaryBilling` or `isPrimaryShipping` param to the `users/save-address` action when saving the current users address. This is mentioned in the https://craftcms.com/docs/commerce/4.x/upgrading.html#saving-an-address
and here https://craftcms.com/docs/commerce/4.x/addresses.html#primary-billing-shipping-addresses

>The (optional) makePrimaryShippingAddress and makePrimaryBillingAddress params are now isPrimaryShipping and isPrimaryBilling.

It was never possible to update the primary address IDs preference on a customer save previously (only on address save or cart address save), but it looks like it should be possible to make it possible in Commerce 4. I will look into that now.

We are working to improve the docs and will take the feedback on board. Thanks!

**Comment by johndwells** [comment]
Thanks @lukeholder , appreciate the steer on this one. I have been able to get this to work as you've outlined.

FWIW, the `users/save-address` endpoint seems to validate all of the addresses' fields, so it's not so simple to fire off a quick POST submission to toggle which address in a list should be set as the primary billing/shipping. The design I am currently working with wants to give the user this ability - e.g. list out all of the addresses saved on file, indicating which one(s) are primary billing/shipping, with a simple toggle button to change. It feels icky having to POST the entire address model just to make a single field change. LMK if you'd like me to raise this as a separate bug or FR.
